src/linalg/exp.rs

```diff
@@ -0,0 +1,479 @@
+            let a6 = self.a6();
+            let ap = &self.a10 as *const Option<MatrixN<N, R>> as *mut Option<MatrixN<N, R>>;
+            unsafe {
+                *ap = Some(a6 * a4);
+            }
+        }
+        self.a10.as_ref().unwrap()
+    }
+
+    fn d4_tight(&mut self) -> N {
+        if self.d4_exact.is_none() {
+            self.d4_exact = Some(one_norm(self.a4()).powf(N::from_f64(0.25).unwrap()));
+        }
+        self.d4_exact.unwrap()
+    }
+
+    fn d6_tight(&mut self) -> N {
+        if self.d6_exact.is_none() {
+            self.d6_exact = Some(one_norm(self.a6()).powf(N::from_f64(1.0 / 6.0).unwrap()));
+        }
+        self.d6_exact.unwrap()
+    }
+
+    fn d8_tight(&mut self) -> N {
+        if self.d8_exact.is_none() {
+            self.d8_exact = Some(one_norm(self.a8()).powf(N::from_f64(1.0 / 8.0).unwrap()));
+        }
+        self.d8_exact.unwrap()
+    }
+
+    fn d10_tight(&mut self) -> N {
+        if self.d10_exact.is_none() {
+            self.d10_exact = Some(one_norm(self.a10()).powf(N::from_f64(1.0 / 10.0).unwrap()));
+        }
+        self.d10_exact.unwrap()
+    }
+
+    fn d4_loose(&mut self) -> N {
+        if self.use_exact_norm {
+            return self.d4_tight();
+        }
+
+        if self.d4_exact.is_some() {
+            return self.d4_exact.unwrap();
+        }
+
+        if self.d4_approx.is_none() {
+            self.d4_approx = Some(one_norm(self.a4()).powf(N::from_f64(0.25).unwrap()));
+        }
+
+        self.d4_approx.unwrap()
+    }
+
+    fn d6_loose(&mut self) -> N {
+        if self.use_exact_norm {
+            return self.d6_tight();
+        }
+
+        if self.d6_exact.is_some() {
+            return self.d6_exact.unwrap();
+        }
+
+        if self.d6_approx.is_none() {
+            self.d6_approx = Some(one_norm(self.a6()).powf(N::from_f64(1.0 / 6.0).unwrap()));
+        }
+
+        self.d6_approx.unwrap()
+    }
+
+    fn d8_loose(&mut self) -> N {
+        if self.use_exact_norm {
+            return self.d8_tight();
+        }
+
+        if self.d8_exact.is_some() {
+            return self.d8_exact.unwrap();
+        }
+
+        if self.d8_approx.is_none() {
+            self.d8_approx = Some(one_norm(self.a8()).powf(N::from_f64(1.0 / 8.0).unwrap()));
+        }
+
+        self.d8_approx.unwrap()
+    }
+
+    fn d10_loose(&mut self) -> N {
+        if self.use_exact_norm {
+            return self.d10_tight();
+        }
+
+        if self.d10_exact.is_some() {
+            return self.d10_exact.unwrap();
+        }
+
+        if self.d10_approx.is_none() {
+            self.d10_approx = Some(one_norm(self.a10()).powf(N::from_f64(1.0 / 10.0).unwrap()));
+        }
+
+        self.d10_approx.unwrap()
+    }
+
+    fn pade3(&mut self) -> (MatrixN<N, R>, MatrixN<N, R>) {
+        let b = [
+            N::from_f64(120.0).unwrap(),
+            N::from_f64(60.0).unwrap(),
+            N::from_f64(12.0).unwrap(),
+            N::from_f64(1.0).unwrap(),
```

Comment:
You should use `crate::convert(120.0)` instead of `N::from_f64` since this is the way we do this everywhere else in nalgebra.

Reply:
Changed everywhere